### PR TITLE
=doc Fix Config links in testing (#25421)

### PR DESCRIPTION
**Pull request by jlprat**
refs: #25421
Fixes links to Configuration html page
Removes link to Ray Roestenburg blog page as it's now returning 404
Fixes link to `ConfigFactory` in Configuration doc page

### CODE REVIEW

File: akka-docs/src/main/paradox/testing.md
Comment (ktoso) [inline akka-docs/src/main/paradox/testing.md]: I've asked @RayRoestenburg about what to do about the link, though I think the drop of the link is perhaps fine... 
Though we could link to the archive post if we wanted to: https://web.archive.org/web/20180114133958/http://roestenburg.agilesquad.com/2011/02/unit-testing-akka-actors-with-testkit_12.html

File: akka-docs/src/main/paradox/testing.md
Comment (ktoso) [inline akka-docs/src/main/paradox/testing.md]: thanks!

**Review by ktoso — APPROVED** [review]
LGTM, pending question to Ray :)

File: akka-docs/src/main/paradox/testing.md
Comment (jlprat) [reply akka-docs/src/main/paradox/testing.md]: Waiting for the reply then 🙂

Comment (jlprat) [reply akka-docs/src/main/paradox/testing.md]: I will rephrase it stating that this is an archive link and add the link to it.

Comment (jlprat) [reply akka-docs/src/main/paradox/testing.md]: Updated

**Review by ktoso — APPROVED** [review]
Looks good, might as well link to the archive I guess ;-)

**Pull request merged**
